### PR TITLE
DBZ-3661 Introduce Oracle `log.mining.archive.destination.name` option

### PR DESCRIPTION
**Pull request by Naros**
https://issues.redhat.com/browse/DBZ-3661

### CODE REVIEW

**Comment by Naros** [comment]
After a quick discussion on Gitter, we concluded the risk is low for this change and that we'll consider adding it to 1.6.

**Review by gunnarmorling — COMMENTED** [review]
LGTM overall; one minor question inline. Thx, @Naros!

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnectorConfig.java [hunk 1]
```diff
@@ -304,6 +304,14 @@
             .withDescription("When set to `false`, the default, LOB fields will not be captured nor emitted. When set to `true`, the connector " +
                     "will capture LOB fields and emit changes for those fields like any other column type.");
 
+    public static final Field LOG_MINING_ARCHIVE_DESTINATION_NAME = Field.create("log.mining.archive.destination.name")
```
[the remark this answers is not in the document]
Comment (Naros) [reply debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnectorConfig.java:307]: Correct, Oracle refers to these as the "Log archive destination(s)" but I don't think I agree with removing the "name" portion.  To me, when saying "Log archive destination" this seems to imply the directory path where they're being read from; however that isn't exactly what those parameter is defining.  In Oracle you can have up to 31 different destinations configured for a variety of reasons.  What is to be provided by this option is the location's configuration name.

For example, I can define:
```
LOG_ARCHIVE_DEST_1    /u01/oracle/app/oradata/redo_logs/
LOG_ARCHIVE_DEST_2    /s01/oradata/long_retention/redo_logs/
```

What the user would provide in this setting would be the actual destination name, or `LOG_ARCHIVE_DEST_n`.  If you don't think removing "name" would give a false impression to users thinking this is the actual path, I'm fine with it but that's mostly my biggest concern is that it _isn't_ the path but the name that points to said path.

[the remark this answers is not in the document]
Comment (jornargelo) [reply debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnectorConfig.java:307]: I found that adding `name ` to the parameter name is actually a good thing. At first I was thinking I had to provide the path to the parameter but the `name` in the parameter made me realize that was not what was wanted. So if my opinion counts for anything I would intuitively say that Chris' original approach would work best for me.

[the remark this answers is not in the document]
Comment (gunnarmorling) [reply debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnectorConfig.java:307]: Yes, I think you're right. In fact, I was assuming it _is_ a path, hence I was stumbling over the "name" part. But given it actually is some symbol which identifies a path, the prefix makes sense. Thanks for clarifying!

**Review by gunnarmorling — APPROVED** [review]

**Comment by gunnarmorling** [comment]
Applied, thx @Naros! One small follow-up question: what happens if the specified destination hasn't been configured in the database before? Perhaps you can push a sentence to the docs to clarifying the requirement that this must exist, or something like that?

**Pull request merged**